src/lib/connectors/db.js

```diff
@@ -1,16 +1,32 @@
-require('dotenv').config();
-const pg = require('pg');
-const moment = require('moment');
-const helpers = require('@envage/water-abstraction-helpers');
+'use strict';
 
-const config = require('../../../config.js');
-const { logger } = require('../../logger');
+const knex = require('./knex');
 
-const DATE_FORMAT = 'YYYY-MM-DD';
-const dateMapper = str => moment(str).format(DATE_FORMAT);
+/**
+ * Allows a query to be run with same arguments
+ * as pg.pool.query(query, params)
+ * The query and params are altered to make them
+ * compatible with knex.raw
+ *
+ * @param {String} sqlQuery - bound params are specified as $1, $2 etc.
+ * @param {Array} [params] - array params
+ * @return {Promise<Object>} query result
+ */
+const query = (sqlQuery, params = []) => {
+  if (params.length === 0) {
+    return knex.knex.raw(sqlQuery);
+  }
 
-// Set dates to format YYYY-MM-DD rather than full date/time string with timezone
-pg.types.setTypeParser(pg.types.builtins.DATE, dateMapper);
+  const data = params.reduce((acc, param, i) => {
+    acc.params[`param_${i}`] = param;
+    const r = new RegExp(`\\$${(i + 1)}(?![0-9])`, 'g');
```

Comment:
I'm not confident I understand what this regex is doing. May I recommend adding an inline comment with a synopsis of its purpose please? I assume it's in some way parsing the query syntax so that it can be passed through knex.raw, but I'm not sure what the transformation is.

Reply:
Thanks I have added comments and tweaked the code slightly